### DISCUSSION TITLE
X96 Mini S905W2: How to install without HDMI

### DISCUSSION THREAD

**Discussion by M0dM** [devmfc/debian-on-amlogic#6] [Q&A]
hi @devmfc and first of all thank you for this awesome repository.

I bought an x96 mini but with s905W2 by mistake and found this repo as my last hope to try to install armbian on this box.
It seems that as soon as the boot from sdcard occurs, the hdmi stops display anything, not even a terminal interface.

I just burned the img on my sdcard and uncommented (as explained in readme.md) the following lines in boot.config : 
box=s905w2_generic
root=/dev/mmcblk0p2

Can I connect to the box via ssh or anything else to complete the setup ?
Did I miss something ?

Or maybe I juste need to wait for better support by the community.

**Comment by devmfc** [comment]
So they are real. I thought they were a scam, but there are really x96 mini's with S905W2's....

It will, with a bit of luck, just boot. 
You can SSH tot the box. But the IP address is not fixed but acquired by DHCP. So you have to scan your own network for it or tell your DHPC server to always give the MAC address of the box a fixed IP.

As a headless server they are quite usable. For HDMI... We'll have to wait I'm afraid.

**Reply by M0dM** [reply]
I bought mine at a discounted price of 27€ on french amazon.

X96 Mini Lecteur Multimédia de Diffusion en continu Android 11.0/Boîte TV 4K avec Amlogic S905W2 Quad-Core chipset, 64 Bits 2GB/16GB,WiFi,4K HD,H.265 by puersit https://amzn.eu/d/bBcE5UO

**Comment by devmfc** [comment]
Maybe you can try one other thing... 
add the following line tot the boot.config:

`bootargs0=pd_ignore_unused clk_ignore_unused`

Maybe this will activate the simple_framebuffer HDMI workaround in the dtb.

**Reply by M0dM** [reply]
Thanks @devmfc !

I will give it a try tomorrow maybe.
I am verry new to the armbian community, what would be the user / password combination to log via ssh ?

**Reply by M0dM** [reply]
![PXL_20231001_144104957](https://github.com/devmfc/debian-on-amlogic/assets/1061506/47e7767c-e6ea-44d0-8aa3-4fa452ed4352)

I got framebuffer working, but it seems that emmc is not working.

And I found the device ip address using nmap but did not managed to log via ssh, maybe the daemon isn't started...

@devmfc Do you have any idea ?

**Reply by devmfc** [reply]
Great to hear that HDMI is sort of working.

Considering your remark about Armbian: my images are not Armbian, are not even Armbian based. I don't have any relationship with them. My images are just plain Debian and Ubuntu bootstrapped images with a for meson64 SOC's patched kernel. 

I don't think the IP address you found is from the tvbox. Because based on your screenshot it is can not find a rootfs and will probably not even have a network connection yet.

To fix this try to change `root=/dev/mmcblk0p2` in boot.config to `root=/dev/mmcblk1p2` I'm afraid I forgot some alias in the DTB for the generic S905W2 box. 
Some other thing you can try instead is to uncomment 'box=tanixw2' instead of' box=s905w2_generic'

username = root password = tvbox

**Reply by M0dM** [reply]
Thanks for the new ideas to dig through

**Comment by M0dM** [comment]
I got it working without having to change to `root=/dev/mmcblk1p2` and `box=tanixw2` but unfortunately only the sdcard is recognized when i do `lsblk` or `fdisk -l`.

I would realy like to make the box booting directly to linux without having to press the hidden button each time...

Do you have any idea ?
By tweaking the dtb file in some ways... maybe ?

**Reply by M0dM** [reply]
I openned the box and it seems EMMC is : THGBMFG7C2LBAIL
Don't know what to do with this information now :roll_eyes:

**Comment by devmfc** [comment]
You can make the sdcard boot every time without pushing the hidden reset button: run ./aml-multiboot-setup.sh, see [readme.md](https://github.com/devmfc/debian-on-amlogic#readme) before you do that..

Why you can't see your eMMC I don't know. The brand of eMMC should not make a difference. Dou you have the boot log (run dmesg)

**Comment by OM-3kChcBjgFR2COfP** [comment]
Good afternoon
I was also lucky to become the owner of the chip s905w2 ( H96Max with 4\32Gb )
When I bought it I didn’t think it would come with it.
Now I don’t know - get rid of it or try this repository?
Did you manage to flash the console properly and use it?

udp:

Nope, won’t start, hangs on the H96Max logo ( with bootargs0=pd_ignore_unused clk_ignore_unused ) or black screen

**Comment by devmfc** [comment]
H96 Max with S905w2? Are you sure?

**Comment by OM-3kChcBjgFR2COfP** [comment]
> H96 Max with S905w2? Are you sure?

![IMG_20231002_085348](https://github.com/devmfc/debian-on-amlogic/assets/20497956/947dfb52-9645-4952-bf64-66e5df5080a5)
![IMG_20231011_134958](https://github.com/devmfc/debian-on-amlogic/assets/20497956/4606492c-6e61-498c-9f90-22752a70964e)
![IMG_20231004_192229](https://github.com/devmfc/debian-on-amlogic/assets/20497956/6b3019ac-09f1-4326-bbe3-3f4691c98733)

**Comment by devmfc** [comment]
Yeah, you're pretty sure ;-)

Just try with burning the image on usb drive and try box=tanixw2 or box=s905w2_generic. If that doesn't work you can try to connect an UART and see what's going on.

But s905w2 is really really experimental and quite slow (especially eMMC)

**Reply by OM-3kChcBjgFR2COfP** [reply]
Then I'll get rid of this console and buy another one. which one is better to take and on which chip? to install armbian\ubuntu?

**Reply by devmfc** [reply]
That depends on what you want? 

Do you want some home server for pi-hole, a mailserver, small webserver or nextcloud like solution? Or just want to have some fun with Linux? Then some S905X3 box with gigabit ethernet will do. I use X96 Max Plus with gigabit ethernet for that. Works great, are cheap and S905X3 has decent mainline support.. But for all of these boxes the Wifi/BT module is a lottery, so don't count on it to work. And even the gigabit phy IC's are a lottery nowadays, but most have mainline support.

Do you expect a good desktop experience with mainline Linux? Buy some Intel/Amd box, it is imho not possible with these Android boxes. But if you really must, buy something with a S922X SoC, like Beelink GT King. But don't expect to have hw accelerated video. 

Do you want Armbian? Don't buy a tvbox, they hate those. And I can't blame them.
Don't like adventures? Or don't want a challenge? Don't buy a tvbox ;-)

**Comment by kerta1n** [comment]
Hey devmfc, I have a Vontair W2 (S905W2 as per the name), and your 6.1.57 build works flawlessly, EXCEPT for eMMC. Like others, it's just not found, but attached is the dmesg output. Something about `EPROBE_DEFER` and error `-110`.


[dmesg.txt](https://github.com/devmfc/debian-on-amlogic/files/12908064/dmesg.txt)
![PXL_20231014_205403241](https://github.com/devmfc/debian-on-amlogic/assets/36344851/0d7a9ac6-d075-41cf-83f1-29633965181e)

**Comment by kerta1n** [comment]
A change in code may be needed:

`... as a follow up, [mmc_hs400_to_hs200()](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/drivers/mmc/core/mmc.c?h=v5.5.13#n1231) is called exclusively by the retuning functionality. Each of the [__mmc_switch()](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/drivers/mmc/core/mmc.c?h=v5.5.13#n1244) calls will likely timeout after ~500ms, so you might be able to infer which step is having the problem quite easily by counting (i.e: HS400 to HS DDR, HS DDR to HS, or HS to HS200). This might be a support request to your supplier. – 
[Attie](https://superuser.com/users/707676/attie)
 [Mar 25, 2020 at 16:11](https://superuser.com/questions/1535420/linux-emmc-mmc-hs400-to-hs200-failed-error-110#comment2330087_1535420)`
 
 (via [superuser.com](https://superuser.com/questions/1535420/linux-emmc-mmc-hs400-to-hs200-failed-error-110) )

**Reply by kerta1n** [reply]
I selected the Tanix W2 box and set the boot partition to `/dev/mmcblk1p2`, and this is the output. I'm going to revert back to `/dev/mmcblk0p2` and see what happens

![PXL_20231014_232410371](https://github.com/devmfc/debian-on-amlogic/assets/36344851/56a58511-74cc-47ed-9a3c-e1d9ae8dbd28)

**Reply by kerta1n** [reply]
`/dev/mmcblk0p2` but also using the Tanix W2 config results in the the system booting, but no eMMC to be found. Same `-110` error.
[dmesg.txt](https://github.com/devmfc/debian-on-amlogic/files/12908210/dmesg.txt)

**Comment by devmfc** [comment]
Can you try attached zipped dtb? (just copy it to _/boot/dtb-something/amlogic_ and set `box=box=s905w2_generic` in _/boot/boot.config_ )

[meson-s4-s905w2-ap201.zip](https://github.com/devmfc/debian-on-amlogic/files/12921422/meson-s4-s905w2-ap201.zip)

**Reply by kerta1n** [reply]
![PXL_20231016_215514305.jpg](https://github.com/devmfc/debian-on-amlogic/assets/36344851/2aa4b4fb-e640-40aa-9b81-d90d29857f17)

That same voltage and -22 error just repeats forever and doesn't boot

**Reply by devmfc** [reply]
I'm afraid that without physical access to a box like yours I can't help you with this. There are a lot of clock/timing differences between the mainline meson_gx_mmc driver and the one from the vendor. For some reason my eMMC on my tanixW2 does work.
Sorry.

**Reply by kerta1n** [reply]
Is it possible for me to fix this by getting root or opening up the box + sending an image?

**Reply by devmfc** [reply]
No that won't help. Problem is not the eMMC module, but probably the combination of the SoC revision + trace lengths + eMMC module and a mainline driver that does not work with that combination. Even with my modifications. To find out hat the problem is you need so many different things to try, it would take ages if you don't have a similar box on your desk.

**Reply by kerta1n** [reply]
Alright. I'll try and crack open the board and see if I can MMC info though, maybe the datasheet might have info about the timings

**Reply by kerta1n** [reply]
I have root access to Android. What files should I be looking for?

**Reply by devmfc** [reply]
There are not really files that  could help you.  It's not like there are some driver files or firmware files, copy them over to mainline Linux and it automagically works... It is really not that simple.
The only thing of a little bit of help is /sys/firmware/fdt, this is the dtb for the vendor kernel, but I think I already have that, but I can could check if something is really different than my s905w2 box.
And maybe a dmesg log (command dmesg or cat /proc/kmsg) might help.

**Reply by kerta1n** [reply]
Ok, the files are sitting at a different house right now, I'll have access to a VPN in around 4 hours. the `/fdt` directory was data only (`file fdt`), and by mistake I thought that because the kernel was booted, I'd only have access to the files in `/devicetree`. I'll probably have access to the box tomorrow. I did end up getting the `dmesg.txt` though, but that is on the same device that needs VPN'ing.

**Reply by kerta1n** [reply]
Ok. I was late by 5 hours, I know, here's dmesg though.
[dmesg.txt](https://github.com/devmfc/debian-on-amlogic/files/13157960/dmesg.txt)

**Reply by kerta1n** [reply]
I got it:
`/devicetree` and the `fdt` file are in here
[files.zip](https://github.com/devmfc/debian-on-amlogic/files/13168809/files.zip)

**Reply by kerta1n** [reply]
[fdtdump.txt](https://github.com/devmfc/debian-on-amlogic/files/13170475/fdtdump.txt)
![Screenshot 2023-10-25 164428](https://github.com/devmfc/debian-on-amlogic/assets/36344851/77f8f75b-ddfe-4455-9597-20d4149358f0)

**Reply by kerta1n** [reply]
I'm going to leave this here to see if this works later
[s905w2-alt.dtb.txt](https://github.com/devmfc/debian-on-amlogic/files/13191459/s905w2-alt.dtb.txt)

DTB files as text:
[meson-s4-s905w2.dts.txt](https://github.com/devmfc/debian-on-amlogic/files/13191562/meson-s4-s905w2.dts.txt)
[weird-android-w2.dts.txt](https://github.com/devmfc/debian-on-amlogic/files/13191563/weird-android-w2.dts.txt)

**Reply by devmfc** [reply]
I didn't find anything in these files that gives any new information. I'm sorry.

**Comment by OM-3kChcBjgFR2COfP** [comment]
thnx for updates, still wait full support s905w2 =)

**Reply by petesimon** [reply]
me too. i got a H96 Max S905W2 for $10 USD after discounts from Aliexpress. Android firmware is now v11, with a date 2023-08-08. I successfully installed Emuelec 4.7beta from _Diegrosan/EmuELEC4.8teste_ from github onto a sd memory card, but I want to try pure Linux. @devmfc

**Comment by kerta1n** [comment]
I got bored and pulled out my Vontar X2 again ([this is what the box looks like](https://www.aliexpress.us/item/3256806317957179.html), labelled "W2 ATV" on the bottom, the board silkscreen is HK1_S905W2_D4_V13, dated 2022-11-30) and decided to see if it would install to eMMC. 

Used `vontar-x2` profile, ran wifi-setup, installed `rsync`, `fdisk`, and `dosfstools` via `apt` manually, then ran multiboot, and finally install-to-emmc. Booted into Debian flawlessly, and now I have another SBC that I could consider throwing Clawdbot on. Thanks again @devmfc 🙌 

Edit: still have to figure out how to get the front LED working, but that is not a priority for me right now

**Reply by kerta1n** [reply]
UPDATE: I was able to get the LED display working by decompling the dtb to a dts, changing the `fd650` section to the following:
```
        fd650 {
                compatible = "devmfc,tm1650";
                dat-gpios = <0x1b 0x47 0x06>;
                clk-gpios = <0x1b 0x45 0x06>;
                segment-mapping = <0x30313233 0x343536ff 0x20212223 0x242526ff 0x10111213 0x141516ff 0x10203 0x40506ff 0xffffffff 0xffffffff 0xffffffff 0xffffffff>;
                status = "okay";
        };
```
backing up the original dtb file, recompiling the PATCHED dts to dtb, moving it back into `boot` and restarting.

Hope this helps and can get included in the newer builds @devmfc

**Reply by devmfc** [reply]
Thanks for reporting back! 
I will add it to the next release.
And I will rename this box to Vontar W2, X2 was a mistake I guess.

One question: you said for installation you manually installed rsync, fdisk. Was that necessary? The script should have doen that for you? Is there a bug in the script that made it fail?

**Reply by kerta1n** [reply]
Yeah the X2 vs W2 can make it confusing because of the listing name, but moving it to W2 would help with a lot for people who are actually holding the device (if possible, a section in the devices list to say something like "the Vontar W2 was probably sold to you as Vontar X2, but the bottom label and box will say W2" [also, the box that I received had "W2 ATV" printed on it as well]).

Yes, the script errored out on either `dosfstools` or `fdisk`. I see the `ensure` line for rsync in the `install-to-emmc` script (but I can't exactly remember why I included `rsync` in the manual list, probably out of habit). There is no `ensure` for `dosfstools` or `fdisk` in the script either, and this is from the 6.18.5 image version. The image doesn't have those utilities installed by default either, and the same goes for utils like `htop`, `file, `lshw`, or `lsusb` (just validated those four as I write this message).

I had one other issue where when I was booted off of the SD card, the WiFi was dropping every 3 minutes; so if I ran the `wifi-connect` script, it would connect, then I would wait a bit to look at documentation, try to install a package after that, with `apt` then failing from a index URL resolve error. (Re-running the wifi-connect script and then the same command IMMEDIATELY after fixes it, so maybe It Was DNS™?)
Now with the OS being on internal eMMC, this problem doesn't seem to be happening anymore.
Not sure if it would be an easy find on your side as to WHY this happened, but I can retry the live image next week, as I will be away for this weekend.

Here are some outputs about info for my WiFi card (looks like BCM4334):
```
tvbox:~#dmesg | grep -iE "brcm|broadcom"
[    4.762290] brcmfmac: brcmf_fw_alloc_request: using brcm/brcmfmac4334-sdio for chip BCM4334/3
[    4.764731] usbcore: registered new interface driver brcmfmac
[    4.766517] brcmfmac mmc2:0001:1: Direct firmware load for brcm/brcmfmac4334-sdio.vontarx2.bin failed with error -2
[    4.781815] brcmfmac mmc2:0001:1: Direct firmware load for brcm/brcmfmac4334-sdio.clm_blob failed with error -2
[    4.992746] brcmfmac: brcmf_c_process_clm_blob: no clm_blob available (err=-2), device may have limited channels available
[    4.992768] brcmfmac: brcmf_c_process_txcap_blob: no txcap_blob available (err=-2)
[    4.993217] brcmfmac: brcmf_c_preinit_dcmds: Firmware: BCM4334/3 wl0: Jul 24 2019 12:58:36 version 6.10.191 (TOB) (r) FWID 01-b049404e
```

```
tvbox:~#cat /sys/class/net/wlan0/device/uevent
DRIVER=brcmfmac
SDIO_CLASS=00
SDIO_ID=02D0:4334
SDIO_REVISION=0.0
MODALIAS=sdio:c00v02D0d4334
```

**Comment by AlexeyDoroshin** [comment]
Hello! 
Vontar w2 with kernel v6.18.21 installed reboots almost every hour. I can't find the reason for these reboots.
[dmesg.txt](https://github.com/user-attachments/files/27207004/dmesg.txt)
